### ISSUE TITLE
Ok I don't want to die writing a framework...so what framework then

### ISSUE THREAD

**Issue by dkebler** [HugoGiraudel/ama#34]
I agree with this tweet you posted

> "You can die writing nice, clean and semantic HTML or you can live long enough to see yourself using a framework."

But I don't want a kitchen sink framework like Foundation.  I tried it and it was impossible to customize as I wished.   I just find buying into a big chunk of other's code unappealing.  You are stuck with their naming conventions, logic, and maybe their mess, documentation or lack thereof, etc.

I have been collecting what I think are excellent small repos of the various elements like typography, modular-scale, media-queries, breakpoints, color handling, buttons, flexbox (as opposed to yet another grid) and have fleshed them out.  I am still poking around looking for a basic set of javascript repos for things like menuing and navigation.    

SO

What I wish for is a skeleton framework to drop them into so I could avoid dying before creating a single website.  A framework that was modular so I could dump the parts I don't like and insert my favourites like above.  In such a sea of frameworks it's time consuming to identify good candidates.

__How is it that you have produced actual websites and yet are still living?__  Do you have already a secret framework you developed as a child that you are not sharing.

**Comment by HugoGiraudel** [comment]
Hey,

I don't have a framework of my own. I don't think I will ever have one, as stated in https://github.com/HugoGiraudel/ama/issues/31. For starters, you have to understand that the few sites I own that I build alone (or mostly) are all running on Jekyll and hosted on GitHub Pages for simplicity concerns (hugogiraudel.com, sass-guidelin.es, sassdoc.com, sass-compatibility.github.io). I didn't need a framework for these, they are all quite small.

Regarding actual projects I work on as part of my full-time job, we usually do not use frameworks either because we want to limit the cluttered mess. It doesn't mean we don't use third-party vendors; just mostly never a full-blown framework. We often rely on a lib for responsive breakpoints (usually include-media as I re-wrote a large part of it), sometimes we use a modular scale but that's quite uncommon.

Long story short, using a CSS framework is tricky if you ask me. Either you need it because you don't want to spend too much time on the CSS side of things, for instance for an admin area of a CMS where Bootstrap is simply perfect (we do that too). Or you need a lot of helpers, and I feel like it's better to rely on a few dependencies that do the job well better than a framework.

If you want to build your collection of tools that you can drop in at the beginning of a project, you could probably create a boilerplate node module (or bower or whatever) that has all the libs you like as dependencies. Then if you want to ditch some things, just have a script do so. Basically automate your project launching with a script.
